lib/shopifyHmac/constructHmacParameter.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const crypto = require('crypto');
+
+const constructHmacParameter = (req, options) =>
+  calculateHmacParam({
+    algorithm: options.getAlgorithm(),
+    sharedSecret: options.sharedSecret,
+    message: getMessage({ ...req }),
+    digest: options.getDigest(),
+  });
+
+const getMessage = ({ query }) => {
+  //   console.log('query', query);
+  const { hmac, ...rest } = query;
+
+  const hash = { ...rest };
+  //   console.log('hash', hash);
+
+  const hashKeys = Object.keys(hash);
+  hashKeys.sort();
+  const message = hashKeys
+    .reduce((accumulator, key) => {
+      if (key === 'ids[]') {
```

Comment:
I am not sure I fully understand what this conditional is for - would you be able to explain it for me?

Reply:
The Shopify Verification docs indicate that an ids array parameter needs to be formatted in a specific way.

> #### IDs array parameter
> If your query string includes an ids array parameter, then the query string is parsed so that `ids[]=1&ids[]=2` gets converted to `ids=["1", "2"]` before the signature is generated.

I've updated the code to work with the Shopify specific example but I need to do more research/testing to see if we need to parse array parameters in the `..ids=1&ids=2...` and `...ids=1,2...` formats.